tests/test_dim_reduction.py

```diff
@@ -0,0 +1,100 @@
+import unittest 
+import pandas as pd 
+import numpy as np 
+import os 
+import sys 
+current_path = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(current_path + "/../neasqc_wp61/models/quantum/")
+from dim_reduction import *
+
+class TestDimReduction(unittest.TestCase):
+    
+    def setUp(self):
+        """
+        Generate random dataset for our tests
+        """
+        n_rows = 10 
+        data = {
+            'label' : np.random.randint(2, size=n_rows),
+            'sentence_embedding' : [
+                np.random.rand(768) for _ in range(n_rows)]
+        }
+        self.df = pd.DataFrame(data)
+
+
+    def testPCA(self):
```

Comment:
This isn't how one names tests.
Tests should have explicit names, ie you should tell what you're testing for. Something like:

PCAOutputsDimensionNEmbeddingsWhenAskedForIt

Like, I should know exactly WHAT the test is testing just by reading the name, without the code or any comments.